libp2p/RLPxHandshake.cpp

```diff
@@ -83,25 +93,43 @@ void RLPXHandshake::readAuth()
 			transition(ec);
 		else if (decryptECIES(m_host->m_alias.sec(), bytesConstRef(&m_authCipher), m_auth))
 		{
-			bytesConstRef sig(&m_auth[0], Signature::size);
-			bytesConstRef hepubk(&m_auth[Signature::size], h256::size);
-			bytesConstRef pubk(&m_auth[Signature::size + h256::size], Public::size);
-			bytesConstRef nonce(&m_auth[Signature::size + h256::size + Public::size], h256::size);
-			pubk.copyTo(m_remote.ref());
-			nonce.copyTo(m_remoteNonce.ref());
-			
-			Secret sharedSecret;
-			crypto::ecdh::agree(m_host->m_alias.sec(), m_remote, sharedSecret);
-			m_remoteEphemeral = recover(*(Signature*)sig.data(), sharedSecret.makeInsecure() ^ m_remoteNonce);
+			bytesConstRef data(&m_auth);
+			auto sig = Signature(data.cropped(0, Signature::size));
```

Comment:
Please change to `Signature sig(data.cropped(0, Signature::size));` (similar below)